src/main/c/Jep/convert_p2j.c

```diff
@@ -702,24 +731,34 @@ static jobject PyCallable_As_JPyCallable(JNIEnv *env, PyObject *pyobject)
     return jpycallable;
 }
 
-static jbyteArray pybufferview_as_jbytearray(JNIEnv *env, Py_buffer view)
-{
-    Py_ssize_t length = view.len / view.itemsize;
-    jbyteArray jarr = (*env)->NewByteArray(env, (jsize) length);
-    if (process_java_exception(env)) {
-        return NULL;
-    }
-    if (PyBuffer_IsContiguous(&view, 'A')) {
-        (*env)->SetByteArrayRegion(env, jarr, 0, (jsize) length, (jbyte*) view.buf);
-    } else {
-        void* buf = (*env)->GetByteArrayElements(env, jarr, NULL);
-        PyBuffer_ToContiguous(buf, &view, view.len, 'C');
-        (*env)->ReleaseByteArrayElements(env, jarr, buf, 0);
-    }
+#ifdef Py_LIMITED_API
+    #define pybufferview_as_array_IMPL_COPY(jtype, Type) \
+        /* NOTE: Our limited wrappers only supported C */ \
+        (*env)->Set##Type##ArrayRegion(env, jarr, 0, (jsize) length, (jtype*) view.buf);
+#else
+    #define pybufferview_as_array_IMPL_COPY(jtype, Type) \
```

Comment:
Although I'm admittedly biased, I feel like these macros are a good thing in general because it significantly reduces code-duplication.